### PR TITLE
When changing view position, y_rot is updated to the nearest multiple of 2*Pi instead of 0, reducing spinning

### PR DESCRIPTION
**Pull request by Nappael**
Extensions to other viewing modes would need to be made as well, as shifting to a side view will still cause spinning.

### CODE REVIEW

**Comment by groud** [comment]
Related #11619

**Comment by akien-mga** [comment]
Your PR contains lots of unrelated commits, you likely did a wrong operation when trying to rebase on the master branch. See http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html for git usage tips :)

**Comment by Nappael** [comment]
Yeah, I noticed that and was worried I had done something wrong. I'll read through this. Sorry about the hassle.

File: editor/plugins/spatial_editor_plugin.cpp [hunk 1]
```diff
@@ -1671,43 +1671,43 @@ void SpatialEditorViewport::_sinput(const Ref<InputEvent> &p_event) {
 			}
 		}
 		if (ED_IS_SHORTCUT("spatial_editor/bottom_view", p_event)) {
-			cursor.y_rot = 0;
+			cursor.y_rot = 2.0 * Math_PI * Math::round(cursor.y_rot / (2.0 * Math_PI));
```
Comment (akien-mga) [inline editor/plugins/spatial_editor_plugin.cpp]: There's a math method to do exactly that: `Math::stepify(cursor.y_rot, 2.0 * Math_PI)`.

**Comment by akien-mga** [comment]
It seems a bit hacky to me, I'd expect the x and y rotations for those pre-defined views to always be in [0; 2PI[ and not to change based on the preview context. Surely there should be another way to prevent spinning in such cases?

**Comment by Nappael** [comment]
The two possible ways of fixing this that I could see were:

1, Modifying the view interpolation code to account for the real angle, not an unbounded number (the cause of the spinning)
2, Modifying the pre-defined view to calculate the nearest required rotation.

1 was preferred, but I couldn't see anything wrong with 2. My logic was that rotating the camera takes you out of that [0:2PI] range anyway, and doing it this way I wouldn't risk interfering with anything else. It is slightly hacky but the logic is there.

I wasn't aware of the stepify function, thank you for that. That makes things easier.

**Pull request closed without merging**